packages/nouns-contracts/contracts/governance/ExcessETH.sol

```diff
@@ -0,0 +1,201 @@
+     */
+
+    uint256 public constant MIN_PAST_AUCTIONS = 2;
+
+    INounsDAOV3 public immutable dao;
+    INounsAuctionHouseV2 public immutable auction;
+    IERC20 public immutable wETH;
+    IERC20 public immutable stETH;
+    IERC20 public immutable rETH;
+    uint256 public immutable waitingPeriodEnd;
+    uint16 public numberOfPastAuctionsForMeanPrice;
+
+    constructor(
+        address owner_,
+        INounsDAOV3 dao_,
+        INounsAuctionHouseV2 auction_,
+        IERC20 wETH_,
+        IERC20 stETH_,
+        IERC20 rETH_,
+        uint256 waitingPeriodEnd_,
+        uint16 numberOfPastAuctionsForMeanPrice_
+    ) {
+        _transferOwnership(owner_);
+
+        dao = dao_;
+        auction = auction_;
+        wETH = wETH_;
+        stETH = stETH_;
+        rETH = rETH_;
+        waitingPeriodEnd = waitingPeriodEnd_;
+        numberOfPastAuctionsForMeanPrice = numberOfPastAuctionsForMeanPrice_;
+    }
+
+    /**
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     *   PUBLIC FUNCTIONS
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     */
+
+    /**
+     * @notice Get the amount of excess ETH in the Nouns treasury.
+     * Excess ETH is defined as the difference between the current treasury value denominated in ETH,
+     * and the expected treasury value denominated in ETH, which is defined as mean auction price * adjusted total supply.
+     * It returns zero during the waiting period, and when expected treasury value is greater than current value.
+     * If treasury balance in native ETH is less than the excess ETH calculation, it returns the treasury ETH balance.
+     * @dev This version does not support burning excess other than in native ETH, e.g. stETH, rETH, etc.
+     * @dev Reverts if there is not enough auction history to calculate the mean auction price.
+     */
+    function excessETH() public view returns (uint256) {
+        if (block.timestamp < waitingPeriodEnd) return 0;
+
+        uint256 expectedTreasuryValue = expectedTreasuryValueInETH();
+        uint256 treasuryValue = treasuryValueInETH();
+
+        if (expectedTreasuryValue >= treasuryValue) return 0;
+
+        return min(treasuryValue - expectedTreasuryValue, owner().balance);
+    }
+
+    /**
+     * @notice Get the expected treasury value denomiated in ETH.
+     * Expected value is defined as mean auction price * adjusted total supply.
+     */
+    function expectedTreasuryValueInETH() public view returns (uint256) {
+        return meanAuctionPrice() * dao.adjustedTotalSupply();
+    }
+
+    /**
+     * @notice Get the current treasury value denomiated in ETH.
+     * In addition to native ETH, it also includes stETH, wETH, and rETH balances.
+     * @dev for rETH, it uses the getEthValue() function to convert rETH to ETH.
+     */
+    function treasuryValueInETH() public view returns (uint256) {
+        address owner_ = owner();
+        return owner_.balance + stETH.balanceOf(owner_) + wETH.balanceOf(owner_) + rETHBalanceInETH();
```

Comment:
@eladmallel nit: feels cleaner to pass `_owner` to `rETHBalanceInETH`

Reply:
thanks, fixed!